doc/plantuml/provisioning_demo_with_keys_and_cert.pu

```diff
@@ -0,0 +1,41 @@
+@startuml
+
+title Demo design for AWS IoT created keys and certificate Provisioning Workflow
+
+skinparam classFontSize 8
+skinparam classFontName Helvetica
+autonumber
+
+box "Demo Application" #LightBlue
+    participant "Demo Main Function" as main
+    participant "<b>KeysAndCertificate</b> Callback" as keysandcert_callback
+    participant "<b>RegisterThing</b> Callback" as registerthing_callback
+end box
+
+box "Provisioning Library" #LightGreen
+    participant "API" as library
+end box
+
+box "Fleet Provisioning feature of\nAWS Iot Core service" #LightPink
+    participant "<b>CreateKeysAndCertificate</b>\n MQTT API" as keysandcertapi
+    participant "<b>RegisterThing</b>\n MQTT API" as registerthingapi
+end box
+
+main -> library: Request new keys and certificate (provide callback)
+note over library: AwsIotProvisioning_CreateKeysAndCertificate
+library -> keysandcertapi: Request new keys and certificate 
+keysandcertapi -> keysandcertapi: Generate new keys and certificate
+keysandcertapi -> library: Send the generated credentials and certificate data.
+library -> keysandcert_callback: Invoke callback with server response
+note left of keysandcert_callback: Print credentials, and \nstore <b>certificate ID</b> and \n<b>certificate ownership token</b>
+
+main -> library: Request registration of device (provide callback)
+note over library: AwsIotProvisioning_RegisterThing
+note left of main: Provides\n1.Certificate ID and Token\n stored in demo application buffers\n2.Template Name (defined in demo config)\n3.Device Parameters (defined in demo app)
+library -> registerthingapi: Request registration of device
+registerthingapi -> registerthingapi: Register <b>certificate</b>,\nattach <b>policy</b>,\ncreate <b>Thing</b> resource,\nand attach <b>certificate to Thing</b>.
+registerthingapi -> library: Send Thing Name,\nand Device Configuration (specified in Fleet Provisioning Template)
+library -> register_thing: Invoke callback with server response
```

Comment:
I think, this has to be library -> registerthing_callback

Reply:
Fixed